Tropos/Models/DailyForecast.swift

```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+@objc class DailyForecast: NSObject {
+    let date: NSDate
+    let conditionsDescription: String
+    let highTemperature: Temperature
+    let lowTemperature: Temperature
+    
+    init(JSON: Dictionary<String, AnyObject>) {
```

Comment:
I do not like failable initializers with classes. The code gets messy so far. You have to fill all stored properties before returning nil. Example for code "improvement".

    @objc class DailyForecast: NSObject {
        let date: NSDate
        let conditionsDescription: String
        let highTemperature: Temperature
        let lowTemperature: Temperature
        
        init?(JSON: Dictionary<String, AnyObject>) {
            if let time = JSON["time"] as? NSTimeInterval {
                self.date = NSDate(timeIntervalSince1970: time)
            } else {
                self.date = NSDate()
            }
            
            if let conditionsDescription = JSON["icon"] as? String {
                self.conditionsDescription = conditionsDescription
            } else {
                self.conditionsDescription = ""
            }
            
            if let temperatureMax = JSON["temperatureMax"] as? Int {
                self.highTemperature = Temperature(fahrenheitValue: temperatureMax)
            } else {
                self.highTemperature = Temperature(fahrenheitValue: 0)
            }
            
            if let temperatureMin = JSON["temperatureMin"] as? Int {
                self.lowTemperature = Temperature(fahrenheitValue: temperatureMin)
            } else {
                self.lowTemperature = Temperature(fahrenheitValue: 0)
            }
        
            super.init()
            if self.dynamicType.validate(JSON) == false { return nil }
        }
        
        class func validate(JSON: Dictionary<String, AnyObject>) -> Bool {
            // json validation goes here
            // check if all necessary values are inside
            return false
        }
    }

What do you think about it @jakecraige? Can we keep it as is for now and think about it more later when we will be able e.g. to convert `DailyForecast` to be struct instead of class? Then we can implement failable intializer even simpler without filling stored properties. It works different for structs.